Added course list page

irinache opened the pull request.

irinache committed:
Added course list page

idle-dt commented on lib/examples_from_courses/presentation/pages/course_list/course_list.dart:
```diff
@@ -0,0 +1,103 @@
+        spacing: 22,
+        children: <Widget>[
+          for (final course in courses)
+            Stack(
```
Вынесите этот Stack в отдельный виджет с названием _CourseListItem. Так будет более читабельный код.

irinache replied:
Сделано.

idle-dt commented on lib/examples_from_courses/presentation/pages/course_list/course_list.dart:
```diff
@@ -0,0 +1,103 @@
+  const _CourseList({Key key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) => Wrap(
```
Почему решили использовать Wrap, а не GridView? К примеру:

GridView.count(
      crossAxisCount: 2,
      children: List.generate(
        6,
        (index) => _CourseListItem(
          title: 'Course $index',
          onTap: (){},
          progress: Percent(50)// instance of the custom Percent class
        ),
      ),
    )

Тогда получится избавиться от SingleChildScrollView выше по дереву.

irinache replied:
Мы просто информацию про Wrap нашли раньше, поэтому его и использовали. Но GridView оказался удобнее, уже заменили на него.

irinache committed:
Updated course_list

- Wrap widget replaced with GridView
- Created a separate widget for course block

irinache committed:
Added small fixes for course block size

idle-dt commented on lib/examples_from_courses/presentation/pages/course_list/course_list.dart:
```diff
@@ -0,0 +1,113 @@
+      );
+}
+
+class Percent extends StatelessWidget {
```
Я имел ввиду сделать тип Percent (Просто класс, который мы будем использовать в моделях для выделения процентов),  а не виджет).

class Percent{
  final double value;

  Percent(this.value);
}



}

А виджет Percent лучше переименовать в PercentWidget.

idle-dt replied:
double value - потому что вдруг будет процент в виде 55,55 и нам надо будет это отображать.

irinache replied:
В новом коммите переделали Percent под тип, а не виджет, и сделали значение double.

irinache committed:
Updated Percent class

- Simple type Percent was made, without inheritance from widget
- Changed way of usage of Percent objects
